Generator/Sources/FileRenderer.swift

```diff
@@ -54,6 +54,36 @@ func renderDocs(_ attribute: Attribute) -> String {
     return result.prefixLines(with: "/// ")
 }
 
+/// Render the `@available` Swift attribute for a deprecated OTel attribute
+///
+/// - Parameters:
+///   - deprecated: The `Deprecated` enum that describes the deprecation
+///   - extendedTypeName: The name of the type being extended with this attribute (i.e. OTelAttribute, or SpanAttributes). Used in the `renamed` argument, if applicable.
+/// - Returns: A string representing the `@available` attribute for Swift
+func renderDeprecatedAttribute(_ deprecated: Deprecated, extendedTypeName: String) -> String {
+    var result = "@available(*, deprecated"
+    switch deprecated {
+    case let .renamed(renamed_to, note):
+        let renamedTo = renamed_to
+        result.append(", renamed: \"\(extendedTypeName).\(renamedTo)\"")
```

Comment:
Here we take the key straight from the OTel semconv deprecation attribute, right? I think this would lead to incorrect fix-its in case the key it was renamed to contained underscores. Those keys would be "swiftified" into `lowerCamelCase` during the generating. Would it be possible to derive these "swiftified" names here and use them to produce correct fix-its in all cases?

Reply:
Ah, excellent point - you're totally right. This is trickier, as we'll probably need it to be dynamic to support possible differences in each renderers' structure. I'll need to move around a few things - I will probably have something back tomorrow.